src/transaction.js

```diff
@@ -72,13 +72,41 @@ export class Transaction extends TransactionBase {
         break;
     }
 
-    const timeBounds = tx.timeBounds();
+    let timeBounds = null;
+    switch (this._envelopeType) {
+      case xdr.EnvelopeType.envelopeTypeTxV0():
+        timeBounds = tx.timeBounds();
+        break;
+
+      case xdr.EnvelopeType.envelopeTypeTx():
+        switch (tx.cond().switch()) {
+          case xdr.PreconditionType.precondTime():
+            timeBounds = tx.cond().timeBounds();
+            break;
+
+          case xdr.PreconditionType.precondV2():
+            timeBounds = tx
+              .cond()
+              .v2()
+              .timeBounds();
+            break;
+
+          default:
+            break;
+        }
+        break;
+
+      default:
+        break;
```

Comment:
‼️I think this default case should throw an error. If we get into the default case it means we either have an invalid tx, or a tx we do not completely understand. A tx we do not completely understand may be particularly dangerous because we may completely misinterpret what the transaction means. (This is one reason why we use a format like XDR where we can explicitly pick and choose what we are okay with being unknown.)